clients/src/main/java/org/apache/kafka/clients/consumer/OffsetAndMetadata.java

```diff
@@ -34,7 +36,12 @@
      */
     public OffsetAndMetadata(long offset, String metadata) {
         this.offset = offset;
-        this.metadata = metadata;
+        // The server converts null metadata to an empty string. So we store it as an empty string as well on the client
```

Comment:
ack. I guess we need never check whether `metadata == null` since no metadata should always mean an empty string. Will update.